addons/web/static/src/core/utils/reactive.js

```diff
@@ -32,21 +32,21 @@ export class Reactive {
  * microtick will only call the original callback once.
  *
  * @param callback the callback to batch
+ * @param synchronize this function decides the granularity of the batch (a microtick by default)
  * @returns a batched version of the original callback
  */
-export function batched(callback) {
+export function batched(callback, synchronize = () => Promise.resolve()) {
     let called = false;
     return async (...args) => {
-        // This await blocks all calls to the callback here, then releases them sequentially
-        // in the next microtick. This line decides the granularity of the batch.
-        await Promise.resolve();
+        // This await blocks all calls to the callback here, then releases them sequentially.
+        await synchronize();
         if (!called) {
             called = true;
             // so that only the first call to the batched function calls the original callback.
             // Schedule this before calling the callback so that calls to the batched function
             // within the callback will proceed only after resetting called to false, and have
             // a chance to execute the callback again
-            Promise.resolve().then(() => (called = false));
+            synchronize().then(() => (called = false));
             callback(...args);
```

Comment:
I think we had a simpler implem of batched, not sure we want to keep this one or have the other